cmd/operator-sdk/add/api.go

```diff
@@ -20,90 +20,261 @@ import (
+	}
+	absProjectPath := projutil.MustGetwd()
+	cfg := &input.Config{
+		AbsProjectPath: absProjectPath,
+	}
+	roleFiles := ansible.RolesFiles{Resource: *r}
+	roleTemplates := ansible.RolesTemplates{Resource: *r}
+
+	// update watch.yaml for the given resource r.
+	if err := ansible.UpdateAnsibleWatchForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the Watch manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+
+	s := &scaffold.Scaffold{}
+	err = s.Execute(cfg,
+		&scaffold.CR{Resource: r},
+		&ansible.RolesReadme{Resource: *r},
+		&ansible.RolesMetaMain{Resource: *r},
+		&roleFiles,
+		&roleTemplates,
+		&ansible.RolesVarsMain{Resource: *r},
+		&ansible.RolesDefaultsMain{Resource: *r},
+		&ansible.RolesTasksMain{Resource: *r},
+		&ansible.RolesHandlersMain{Resource: *r},
+	)
+	if err != nil {
+		log.Fatalf("API scaffold failed: %v", err)
+	}
+	if err = genutil.GenerateCRDNonGo("", *r, apiFlags.CrdVersion); err != nil {
+		return err
+	}
+	// update deploy/role.yaml for the given resource r.
+	if err := scaffold.UpdateRoleForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the RBAC manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+	return nil
+
+}
+
+// TODO
+// Consolidate scaffold func to be used by both "new" and "add api" commands.
+func doHelmAPIScaffold() error {
+
+	absProjectPath := projutil.MustGetwd()
+	projectName := filepath.Base(absProjectPath)
+	cfg := &input.Config{
+		AbsProjectPath: absProjectPath,
+		ProjectName:    projectName,
+	}
+
+	createOpts := helm.CreateChartOptions{
+		ResourceAPIVersion: apiFlags.APIVersion,
+		ResourceKind:       apiFlags.Kind,
+		Chart:              apiFlags.HelmChartRef,
+		Version:            apiFlags.HelmChartVersion,
+		Repo:               apiFlags.HelmChartRepo,
+	}
+
+	r, chart, err := helm.CreateChart(cfg.AbsProjectPath, createOpts)
+	if err != nil {
+		return fmt.Errorf("failed to create helm chart: %v", err)
+	}
+
+	valuesPath := filepath.Join("<project_dir>", helm.HelmChartsDir, chart.Name(), "values.yaml")
+
+	rawValues, err := yaml.Marshal(chart.Values)
+	if err != nil {
+		return fmt.Errorf("failed to get raw chart values: %v", err)
+	}
+	crSpec := fmt.Sprintf("# Default values copied from %s\n\n%s", valuesPath, rawValues)
+
+	// update watch.yaml for the given resource r.
+	if err := helm.UpdateHelmWatchForResource(r, absProjectPath); err != nil {
+		return fmt.Errorf("failed to update the Watch manifest for the resource (%v, %v): (%v)",
+			r.APIVersion, r.Kind, err)
+	}
+
+	s := &scaffold.Scaffold{}
+	err = s.Execute(cfg,
+		&scaffold.CR{
+			Resource: r,
+			Spec:     crSpec,
+		},
+	)
+	if err != nil {
+		log.Fatalf("API scaffold failed: %v", err)
+	}
+	if err = genutil.GenerateCRDNonGo("", *r, apiFlags.CrdVersion); err != nil {
 		return err
 	}
 
-	log.Infof("Generating api version %s for kind %s.", apiVersion, kind)
+	roleScaffold := helm.DefaultRoleScaffold
+
```

Comment:
See that after run `operator-sdk add api --api-version=cache.example.com/v1alpha1 --kind=AppService` on the Ansible Memcached Sample no changes would need to be required to be done and the resource is added after '*'

<img width="1350" alt="Screenshot 2020-04-09 at 18 34 40" src="https://user-images.githubusercontent.com/7708031/78924048-13d82380-7a91-11ea-9db6-004e4ed12f9a.png">

However, it is working 100% well when a new group is used. E.g `operator-sdk add api --api-version=test.example.com/v1alpha1 --kind=Test
`
<img width="1372" alt="Screenshot 2020-04-09 at 18 39 06" src="https://user-images.githubusercontent.com/7708031/78924280-6ca7bc00-7a91-11ea-9074-d58c89a3b4a5.png">

Reply:
This is how it does, even for existing GO .